internal/reporter/downstream_reporter/cache/cache.go

```diff
@@ -85,7 +68,43 @@ func (cacheReporter *Cache) addExecution(newExecutionEntry cache_report.Executio
 	return nil
 }
 
+func (cacheReporter *Cache) GetExecutions() ([]cache_report.ExecutionEntry, error) {
+	cacheReporter.mutex.Lock()
```

Comment:
I would gaurd the most critical sections in getExecutions and addExecutions. Also then just use a go routine to let the decomposer not wait on any mutex.